File: Iot_raspberrypi/01_gpio_digital/led_blink_test.py
import RPi.GPIO as GPIO
import time

LED_PIN_r = 4
LED_PIN_y = 5
LED_PIN_g = 6

GPIO.setmode(GPIO.BCM) #BCM pin번호를 따르겠다는 의미
GPIO.setup(LED_PIN_r, GPIO.OUT)
GPIO.setup(LED_PIN_y, GPIO.OUT)
GPIO.setup(LED_PIN_g, GPIO.OUT)

'''
GPIO.output(LED_PIN_g, GPIO.LOW)
GPIO.output(LED_PIN_r, GPIO.LOW)
GPIO.output(LED_PIN_y, GPIO.LOW)
'''

GPIO.output(LED_PIN_r, GPIO.HIGH)
print("led RED on")

time.sleep(2)

GPIO.output(LED_PIN_r, GPIO.LOW)
GPIO.output(LED_PIN_y, GPIO.HIGH)
print("led RED off and led YELLOW on")

time.sleep(2)

GPIO.output(LED_PIN_y, GPIO.LOW)
GPIO.output(LED_PIN_g, GPIO.HIGH)
print("led YELLOW off and led GREEN on")

time.sleep(2)

GPIO.output(LED_PIN_g, GPIO.LOW)
print("led GREEN off")


GPIO.setup(LED_PIN_r, GPIO.IN)
GPIO.setup(LED_PIN_y, GPIO.IN)
GPIO.setup(LED_PIN_g, GPIO.IN)
GPIO.cleanup() #pin 초기화

File: Iot_raspberrypi/01_gpio_digital/led_blink_test.c
#include <wiringPi.h>
#include <stdio.h>
#define LED_r 4
#define LED_g 5
#define LED_y 6

int main (void)
{
  //wiringPiSetup () ; //physical pin번호로 할때
  wiringPiSetupGpio(); //bcm pin번호로 할때
  pinMode (LED_r, OUTPUT);
  pinMode (LED_g, OUTPUT);
  pinMode (LED_y, OUTPUT);
  
  digitalWrite (LED_r, HIGH);
  printf("led RED on\n");
  delay (2000);
  digitalWrite (LED_r,  LOW);
  digitalWrite (LED_g, HIGH);
  printf("led RED off and led GREEN on\n");
  delay (2000);
  digitalWrite (LED_g,  LOW);
  digitalWrite (LED_y, HIGH);
  printf("led GREEN off and led YELLOW on\n");
  delay (2000);
  digitalWrite (LED_y, LOW);
  printf("led YELLOW off\n");
  
  pinMode (LED_r, INPUT);
  pinMode (LED_g, INPUT);
  pinMode (LED_y, INPUT);
  
  return 0 ; 
}